### PR TITLE
Adding validation_steps and batch_size to VaidationMonitor

### PR DESCRIPTION
**Pull request by ilblackdragon**
This allows to do less frequent and mini batch evaluation while training.

Ref https://github.com/tensorflow/skflow/issues/133

### CODE REVIEW

File: tensorflow/contrib/learn/python/learn/monitors.py [hunk 1]
```diff
@@ -157,15 +157,20 @@ class ValidationMonitor(BaseMonitor):
         early_stopping_rounds:  Activates early stopping if this is not None.
                                 Loss needs to decrease at least every <early_stopping_rounds>
                                 round(s) to continue training. (default: None)
+        validation_steps: Number of training steps to run validation between.
+                          If None, print_steps is used.
```
Comment (martinwicke) [inline tensorflow/contrib/learn/python/learn/monitors.py]: batch_size doc is missing

File: tensorflow/contrib/learn/python/learn/monitors.py [hunk 2]
```diff
@@ -177,14 +182,26 @@ def set_estimator(self, estimator):
     def create_val_feed_dict(self, inp, out):
         """Set tensorflow placeholders and create validation data feed"""
         self.val_feeder.set_placeholders(inp, out)
-        self.val_dict = self.val_feeder.get_feed_dict_fn()()
+        # If mini batches, get a function else full dataset.
+        if self._batch_size > 0:
+            self.val_dict_fn = self.val_feeder.get_feed_dict_fn()
+        else:
+            self.val_dict = self.val_feeder.get_feed_dict_fn()()
```
Comment (martinwicke) [inline tensorflow/contrib/learn/python/learn/monitors.py]: Why the distinction? Together with the branch in _set_last_loss_seen which always does the same thing in the end -- compute self.val_feeder.get_feed_dict_fn()(). 

Is val_dict used elsewhere?

**Comment by vrv** [comment]
thar be conflicts, please update!

**Comment by martinwicke** [comment]
Ping @ilblackdragon. Is this still relevant?

**Pull request closed without merging**